[Greg-Tan] iP

Greg-Tan opened the pull request:
### HughJazz

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

HughJazz frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [X]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class HughJazz {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Greg-Tan committed:
rename, greet, exit

Greg-Tan committed:
Echo

Greg-Tan committed:
Add, List

Greg-Tan committed:
Mark as Done

Greg-Tan committed:
Todos, Events, Deadlines

Greg-Tan committed:
Automated Text UI Testing

Greg-Tan committed:
Handle Errors

Greg-Tan committed:
Delete

Greg-Tan committed:
Save

cyqjoseph commented on src/main/java/HughJazz.java:
```diff
@@ -0,0 +1,189 @@
+    }
+
+    private static void handleInput(String userInput) throws ChatbotException {
+        if (userInput.toLowerCase().startsWith("todo")) {
```
An alternative is to switch/case statements here?

cyqjoseph commented on src/main/java/HughJazz.java:
```diff
@@ -0,0 +1,189 @@
+import java.io.PrintWriter;
+import java.io.IOException;
+public class HughJazz {
+    private static final int MAX_TASKS = 100;
```
Great use of `MAX_TASKS` as a constant!

cyqjoseph commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,18 @@
+public class Todo extends Task {
```
An alternative could be to use switch case statements here?

cyqjoseph commented on src/main/java/HughJazz.java:
```diff
@@ -0,0 +1,189 @@
+import java.io.File;
```
Consider keeping imports in alphabetical order!

cyqjoseph reviewed (commented):
Overall, the code generaly follows good style practices. Ensure consistent indentation and spacing for better readability between your function definitions. Some of the methods such as `addTask` are also quite long. Consider abstracting out certain functionalities of `addTask` to another functions!

Greg-Tan committed:
Changed data txt file

Greg-Tan committed:
Dates and Times

Greg-Tan committed:
Merge branch 'branch-Level-8'

Greg-Tan committed:
More OOP

Greg-Tan committed:
Updated code for More OOP

Greg-Tan committed:
Organised classes into one package

Greg-Tan committed:
Merge branch 'branch-A-Packages'

Greg-Tan committed:
Merge branch 'add-gradle-support'

Greg-Tan committed:
Use Gradle

Greg-Tan committed:
Added JUnit tests

Greg-Tan committed:
Added new JUnit tests for Todo and Deadline

Greg-Tan committed:
Merge branch 'branch-A-JUnit'

Greg-Tan committed:
updated gitignore to ignore the META-INF file

Greg-Tan committed:
Updated gitignore to exclude META-INF

Greg-Tan committed:
Added Javadoc comments to the classes

Greg-Tan committed:
Improve Parser class to include more abstraction

Greg-Tan committed:
Change code format to fit Java Coding Standard

Greg-Tan committed:
Implement find functionality

Greg-Tan committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/HughJazz.java
#	src/main/java/duke/Parser.java

Greg-Tan committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/HughJazz.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java

Greg-Tan committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/TaskList.java

Greg-Tan committed:
resolve issues with file extensions

Greg-Tan committed:
added GUI components

Greg-Tan committed:
Edit code to improve functionality and features

Greg-Tan committed:
Merge branch 'branch-Level-10'

Greg-Tan committed:
Add assertions to document internal assumptions

- Introduced assertions to ensure the 'tasks' list is never null before performing operations like addTask, deleteTask, and markTask. This helps catch any initialization issues early in the development phase.
- Added assertions in 'deleteTask' and 'markTask' methods to verify that task numbers are within the valid range before proceeding. This is to document the assumption that these methods are called with valid task numbers, reflecting internal logic expectations.
- Asserted that descriptions of tasks being added are not null or empty, reinforcing the precondition that tasks must have meaningful descriptions.
- These assertions are intended for use during development and debugging to identify logical errors related to task list management more quickly.

Greg-Tan committed:
Improve code quality

- Added extra break lines to structure the code more logically

- Refactored complicated expressions into simpler ones

- Removed magic strings

Greg-Tan committed:
Merge pull request #2 from Greg-Tan/branch-A-CodeQuality

Improve code quality

Greg-Tan committed:
Merge branch 'master' into branch-A-Assertions

Greg-Tan committed:
Merge pull request #3 from Greg-Tan/branch-A-Assertions

Add assertions to document internal assumptions

Greg-Tan committed:
Fix bug in the code

Greg-Tan committed:
Resolve dateTime formatting issue

Greg-Tan committed:
Add functionality to detect duplicates

Bot checks current task list and throws an exception if the input matches the description of a task already existing in the task list, preventing duplicate entries.

Greg-Tan committed:
Fix checkstyle

Greg-Tan committed:
Merge branch 'branch-C-DetectDuplicates'

# Conflicts:
#	src/main/java/duke/TaskList.java

Greg-Tan committed:
Publish User Guide and Screenshots

Greg-Tan committed:
Fix formatting issues
